gamification-evm-services/src/main/java/io/meeds/evm/gamification/listener/EvmRuleUpdateListener.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * This file is part of the Meeds project (https://meeds.io/).
+ *
+ * Copyright (C) 2020 - 2024 Meeds Association contact@meeds.io
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package io.meeds.evm.gamification.listener;
+
+import io.meeds.evm.gamification.utils.Utils;
+import io.meeds.gamification.model.RuleDTO;
+import io.meeds.gamification.service.RuleService;
+import jakarta.annotation.PostConstruct;
+import org.exoplatform.commons.api.persistence.ExoTransactional;
+import org.exoplatform.commons.exception.ObjectNotFoundException;
+import org.exoplatform.services.listener.Event;
+import org.exoplatform.services.listener.Listener;
+
+import org.exoplatform.services.listener.ListenerService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.List;
+
+import static io.meeds.evm.gamification.utils.Utils.EVM_SAVE_ACTION_EVENT;
+
+@Service
```

Comment:
Not a service, but a plugin, thus use `@Component` instead

```suggestion
@Component
```